fabtests/benchmarks/rdm_bw_mt.c

```diff
@@ -0,0 +1,767 @@
+		if (ret != -FI_EAGAIN)
+			return ret;
+
+		force_progress(targs->cq);
+	} while (1);
+}
+
+static int post_recv(void *context)
+{
+	int ret;
+	struct thread_args *targs = context;
+
+	do {
+		ret = fi_recv(targs->ep, targs->rx_buf, targs->size,
+			      targs->rx_mr, targs->fiaddr, &targs->recv_ctx);
+		if (ret != -FI_EAGAIN)
+			return ret;
+
+		printf("%d fi_recv failed: %d\n", targs->id, ret);
+		force_progress(targs->cq);
+	} while (1);
+}
+
+static int bw_send(void *context)
+{
+	int ret;
+	struct thread_args *targs = context;
+
+	ret = post_send(context);
+	if (ret)
+		return ret;
+	ret = read_cq(targs->cq);
+	if (ret) {
+		printf("send read_cq error: %d\n", ret);
+		return ret;
+	}
+	return 0;
+}
+
+static int bw_recv(void *context)
+{
+	int ret = FI_SUCCESS;
+	struct thread_args *targs = context;
+
+	ret = post_recv(context);
+	if (ret)
+		return ret;
+
+	ret = read_cq(targs->cq);
+	if (ret) {
+		printf("recv read_cq error: %d\n", ret);
+		return ret;
+	}
+
+	return 0;
+}
+
+static void *uni_bandwidth(void *context)
+{
+	int i, ret;
+
+	pthread_barrier_wait(&barrier);
+	for (i = 0; i < opts.warmup_iterations; i++) {
+		ret = opts.dst_addr ? bw_send(context) : bw_recv(context);
+		if (ret) {
+			((struct thread_args *) context)->ret = ret;
+			printf("ep[%d] warmup failed iter %d\n", targs->id, i);
+			break;
+		}
+	}
+
+	pthread_barrier_wait(&barrier);
+	if (targs->id == 0)
+		ft_start();
+	for (i = 0; i < opts.iterations; i++) {
+		ret = opts.dst_addr ? bw_send(context) : bw_recv(context);
+		if (ret) {
+			((struct thread_args *) context)->ret = ret;
+			break;
+		}
+	}
+	pthread_barrier_wait(&barrier);
+	if (targs->id == 0)
+		ft_stop();
+
+	return  NULL;
+}
+
+static void *bi_bandwidth(void *context)
+{
+	int i, ret;
+	struct thread_args *targs = context;
+
+	pthread_barrier_wait(&barrier);
+	for (i = 0; i < opts.warmup_iterations; i++) {
+		ret = opts.dst_addr ? bw_send(context) : bw_recv(context);
+		if (ret) {
+			((struct thread_args *) context)->ret = ret;
+			printf("%d warmup failed iter %d\n", targs->id, i);
+			break;
+		}
+		ret = opts.dst_addr ? bw_recv(context) : bw_send(context);
+		if (ret) {
+			((struct thread_args *) context)->ret = ret;
+			printf("%d warmup failed iter %d\n", targs->id, i);
+			break;
+		}
+	}
+
+	pthread_barrier_wait(&barrier);
+	if (targs->id == 0)
+		ft_start();
+
+	for (i = 0; i < opts.iterations; i++) {
+		ret = opts.dst_addr ? bw_send(context) : bw_recv(context);
+		if (ret) {
+			((struct thread_args *) context)->ret = ret;
+			printf("%d warmup failed iter %d\n", targs->id, i);
+			break;
+		}
+		ret = opts.dst_addr ? bw_recv(context) : bw_send(context);
+		if (ret) {
+			((struct thread_args *) context)->ret = ret;
+			printf("%d warmup failed iter %d\n", targs->id, i);
+			break;
+		}
+	}
+	pthread_barrier_wait(&barrier);
+	if (targs->id == 0)
+		ft_stop();
+
+	return NULL;
+}
+
+static int run_size(size_t size)
+{
+	int ret = FI_SUCCESS;
+	int err = FI_SUCCESS;
```

Comment:
No need to initialize err, consolidate on one line
int err, i, ret = FI_SUCCESS;